plugins/integration/lib/events/kIntegrationFlowManager.php

```diff
@@ -59,7 +70,40 @@ public static function addintegrationJob($objectType, $objectId, kIntegrationJob
 		$batchJob->setStatus(BatchJob::BATCHJOB_STATUS_DONT_PROCESS);
 		
 		$jobType = IntegrationPlugin::getBatchJobTypeCoreValue(IntegrationBatchJobType::INTEGRATION);
-		$batchJob = kJobsManager::addJob($batchJob, $data, $jobType, $data->getProviderType());
+		$batchJob = kJobsManager::addJob($batchJob, $data, $jobType, $providerType);
+		
+		if($integrationProviderObj && $integrationProviderObj->shouldSendCallBack())
+		{
+			$ks = self::generateKs($partnerId, $batchJob->getId());
+			$data = $batchJob->getData();
+			$data->setKsForExternalService($ks);
```

Comment:
Great idea, you can take it further, you could already set the callbackNotificationUrl, then you won't have to calculate it in KalturaIntegrationJobData constructor and you won't have to append the job id id in KAsyncIntegrate::getCallbackNotificationUrl and you can even rename callbackNotificationBaseUrl to callbackNotificationUrl.